mlflow/projects/utils.py

```diff
@@ -29,8 +29,7 @@
 )
 from mlflow.utils.rest_utils import augmented_raise_for_status
 
-# TODO: this should be restricted to just Git repos and not S3 and stuff like that
-_GIT_URI_REGEX = re.compile(r"^[^/]*:")
+_GIT_URI_REGEX = re.compile(r"((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?")
```

Comment:
Can we add a test (e.g. `test_git_uri_regex`) in `tests/projects/test_utils.py` to ensure this doesn't match a windows path?